bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/internal/servlets/ReviewDataSourceServlet.java

```diff
@@ -0,0 +1,134 @@
+import org.apache.sling.api.resource.SyntheticResource;
+import org.apache.sling.api.resource.ValueMap;
+import org.apache.sling.api.wrappers.ValueMapDecorator;
+import org.jetbrains.annotations.NotNull;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.adobe.cq.export.json.ComponentExporter;
+import com.adobe.cq.forms.core.components.internal.form.FormConstants;
+import com.adobe.cq.forms.core.components.models.form.Base;
+import com.adobe.cq.forms.core.components.models.form.Container;
+import com.adobe.cq.forms.core.components.models.form.FormComponent;
+import com.adobe.cq.forms.core.components.models.form.FormContainer;
+import com.adobe.cq.forms.core.components.models.form.Label;
+import com.adobe.cq.forms.core.components.util.ComponentUtils;
+import com.adobe.granite.ui.components.ExpressionResolver;
+import com.adobe.granite.ui.components.ds.DataSource;
+import com.adobe.granite.ui.components.ds.SimpleDataSource;
+import com.adobe.granite.ui.components.ds.ValueMapResource;
+
+@Component(
+    service = { Servlet.class },
+    property = {
+        "sling.servlet.resourceTypes=" + FormConstants.RT_FD_FORM_REVIEW_DATASOURCE_V1,
+        "sling.servlet.methods=GET",
+        "sling.servlet.extensions=html"
+    })
+public class ReviewDataSourceServlet extends AbstractDataSourceServlet {
+
+    /**
+     * Defines the form meta data type. Possible values: {@code submitAction},
+     * {@code prefillServiceProvider}
+     *
+     * @todo: Add other metadata types here like fragment, actions etc
+     */
+    @Reference
+    private transient ExpressionResolver expressionResolver;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ReviewDataSourceServlet.class);
+
+    @NotNull
+    @Override
+    protected ExpressionResolver getExpressionResolver() {
+        return expressionResolver;
+    }
+
+    @Override
+    protected void doGet(@NotNull SlingHttpServletRequest request, @NotNull SlingHttpServletResponse response) {
+
+        ResourceResolver resourceResolver = request.getResourceResolver();
+        String componentInstancePath = request.getRequestPathInfo().getSuffix();
+        List<Resource> resources = new ArrayList<>();
+        if (resourceResolver != null) {
+            Resource componentInstance = resourceResolver.getResource(componentInstancePath);
+            Resource formInstance = ComponentUtils.getFormContainer(componentInstance);
+            FormContainer formContainer = formInstance.adaptTo(FormContainer.class);
+            List<Base> panels = (List<Base>) getPanels(formContainer);
+            for (Base panel : panels) {
+                String id = panel != null ? panel.getId() : "";
+                String title = "";
+                if (panel != null) {
+                    Label label = panel.getLabel();
+                    if (label != null) {
```

Comment:
we can also write if (label != null && label.getValue() != null) {
title = value;
}